### PR TITLE
[CP-2995] Introduce adjustedOffset logic to handle tooltip flipping adjustments

### PR DESCRIPTION
**Pull request by dkarski**
JIRA Reference: [CP-2995]

### :memo: Description ️

Refactor tooltip positioning logic by introducing adjustedOffset to handle flipping adjustments for more consistent and maintainable offset calculations.

```


// @ts-ignore
const view: View = {
  phoneDropdownCounterBadge: {
    component: "button-text",
    config: {
      actions: [],
      modifiers: ["hover-background"],
    },
    layout: {
      padding: "2px 5px",
    },
    dataProvider: {
      source: "entities-field",
      entitiesType: "contacts",
      fields: [
        {
          slice: [1],
          flat: "phoneNumber",
          providerField: "phoneNumbers",
          componentField: "extra-data.tooltip.contentList",
        },
      ],
    },
    extra: {
      tooltip: {
        placement: "bottom-left",
        offset: {
          x: 0,
          y: 15,
        },
      },
    },
    childrenKeys: ["phoneDropdownCounterBadgeText"],
  },
  columnCheckbox: {
    component: "table.cell",
    config: {
      width: "74",
    },
    layout: {
      padding: "0 0 0 32px",
    },
    childrenKeys: ["contactCheckbox"],
  },
  contactCheckbox: {
    component: "form.checkboxInput",
    config: {
      name: "selectedContacts",
      size: "small",
    },
    dataProvider: {
      source: "entities-field",
      entitiesType: "contacts",
      fields: [
        {
          providerField: "contactId",
          componentField: "config.value",
        },
      ],
    },
    extra: {
      tooltip: {
        contentText: "Select",
        offset: {
          x: 0,
          y: 15,
        },
      },
    },
  },
}

export default view
```

<details>

Zrzut ekranu 2024-10-4 o 10 58 48
Zrzut ekranu 2024-10-4 o 10 58 46
Zrzut ekranu 2024-10-4 o 10 21 45
Zrzut ekranu 2024-10-4 o 10 58 41

</details>

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2995]: https://appnroll.atlassian.net/browse/CP-2995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx [hunk 1]
```diff
@@ -59,30 +61,47 @@ export const Tooltip: BaseGenericComponent<
         let top = 0
         let left = 0
 
+        const adjustedOffset: TooltipOffsetType = { ...offset }
+        const { isFlippedVertically, isFlippedHorizontally } = getFlipStatus(
+          originalPlacement.current,
+          placment
```
Comment (mkurczewski) [inline libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx:67]: Should be `placement`

**Review by mkurczewski — COMMENTED** [review]
Consider extracting `calculatePosition`  and `isWithinViewport` functions out of `handleAnchorHover` scope.

File: libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx (hunk 1 above)
Comment (dkarski) [reply libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx]: https://github.com/mudita/mudita-center/pull/2110/commits/845c79eba96c8c57fab1cb71f7302addb71776b0

File: libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx [hunk 2]
```diff
@@ -59,30 +61,47 @@ export const Tooltip: BaseGenericComponent<
...
+              anchorRect.left -
+              contentRect.width +
+              anchorRect.width +
+              adjustedOffset.x
             break
           default:
-            top = anchorRect.top + anchorRect.height + offset.y
-            left = anchorRect.left + offset.x
+            top = anchorRect.top + anchorRect.height + adjustedOffset.y
+            left = anchorRect.left + adjustedOffset.x
```
[the remark this answers is not in the document]
Comment (dkarski) [reply libs/generic-view/ui/src/lib/interactive/tooltip/tooltip.tsx]: https://github.com/mudita/mudita-center/pull/2110/commits/845c79eba96c8c57fab1cb71f7302addb71776b0

**Review by mkurczewski — APPROVED** [review]

**Pull request merged**